ts_src/psbt.ts

```diff
@@ -642,14 +665,31 @@ export class Psbt {
       sighashTypes,
     );
 
-    const partialSig = [
-      {
+    const scriptType = this.getInputType(inputIndex);
+
+    if (isTaprootSpend(scriptType)) {
+      if (!keyPair.signSchnorr) {
+        throw new Error(
+          `Need Schnorr Signer to sign taproot input #${inputIndex}.`,
+        );
+      }
+      const partialSig = this.data.inputs[inputIndex].partialSig || [];
```

Comment:
`PartialSig` 0x02 can't be reused for Taproot -- we need to implement `TapScriptSig` 0x14 :(

`PartialSig` is specifically an XY pubkey and ECDSA signature.

This problem shows up when trying to improve the `test/psbt.utils.ts` `buildTapscriptFinalizer` to properly sequence signatures to match the script, and also would come up if a PSBT happened to have signatures corresponding to potentially multiple different spend paths.

Reply:
bip174 library is pretty modular so adding TapScriptSig should be trivial.

Reply:
Cool. I'm working on adding all of the BIP371 extensions (they're part of BIP174 since January, so this seems appropriate).